backend/src/main/java/gov/cdc/usds/simplereport/service/TestOrderService.java

```diff
@@ -202,6 +213,59 @@ public int cancelAll() {
 	  return _repo.cancelAllPendingOrders(_os.getCurrentOrganization());
   }
 
+    @Transactional
+    @AuthorizationConfiguration.RequirePermissionUpdateTest
+    public TestEvent correctTestMarkAsError(String testEventIdStr, String reasonForCorrection) {
+        // The client sends us a TestEvent, we need to map back to the Order.
+        UUID testEventId = UUID.fromString(testEventIdStr);
+        Optional<TestEvent> loadExistingEvent = _terepo.findById(testEventId);
+        if (loadExistingEvent.isEmpty()) {
+            // should this throw?
+            LOG.error("Failed to load TestEvent by id {}", testEventId);
+            return null;
+        }
```

Comment:
Yes I think this should throw an error
```suggestion
        TestEvent loadExistingEvent = _terepo.findById(testEventId).orElseThrow(()->new IllegalGraphqlArgumentException("invalid test event ID"));
```